### PR TITLE
Allow specification of controller names for trajectories

### PR DESCRIPTION
**Pull request by llach**
### Description

This PR adds a list of controller names to `ExecutableTrajectory` which is passed to the `TrajectoryExecutionManager` (TEM) in `PlanExecution::executeAndMonitor()`. This allows users to specify which controllers should execute a certain trajectory (or only a part of it). The TEM already had the ability to manage different controllers for different trajectory parts, it just was not used before.

Additonally, the `FakeControllerManager` now reads the `default` parameter from the `controller_list` on the param server and initializes the `ControllerState`  accordingly. Before, each `ControllerState` had `default_` and `active_` set to `true` which lead the TEM's controller selection mechanism to choose any controller when no explicit controller name was given.
[The documentation](https://ros-planning.github.io/moveit_tutorials/doc/controller_configuration/controller_configuration_tutorial.html) mentions this parameter, but neither [SimpleControllerManager](https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp#L227) nor [FakeControllerManager](https://github.com/ros-planning/moveit/blob/master/moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp#L265) use it.

When I did these changes, I stumbled upon comments hinting that controller switching was possible or discontinued. Someone with a better overview of MoveIt should probably decide whether it okay to allow it again. 

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the MIGRATION.md notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning/plan_execution/include/moveit/plan_execution/plan_representation.h [hunk 1]
```diff
@@ -52,8 +52,12 @@ struct ExecutableTrajectory
   {
   }
 
-  ExecutableTrajectory(const robot_trajectory::RobotTrajectoryPtr& trajectory, const std::string& description)
-    : trajectory_(trajectory), description_(description), trajectory_monitoring_(true)
+  ExecutableTrajectory(const robot_trajectory::RobotTrajectoryPtr& trajectory, const std::string& description,
+                       std::vector<std::string> controller_names = std::vector<std::string>(1, ""))
```
Comment (rhaschke) [inline moveit_ros/planning/plan_execution/include/moveit/plan_execution/plan_representation.h]:
```suggestion
                       std::vector<std::string> controller_names = {""})
```

File: moveit_ros/planning/plan_execution/src/plan_execution.cpp [hunk 2]
```diff
@@ -370,7 +370,9 @@ moveit_msgs::MoveItErrorCodes plan_execution::PlanExecution::executeAndMonitor(E
     // convert to message, pass along
     moveit_msgs::RobotTrajectory msg;
     plan.plan_components_[i].trajectory_->getRobotTrajectoryMsg(msg);
-    if (!trajectory_execution_manager_->push(msg))
+
+    // trajectory execution manager handles empty controller name vectors in its configure(), so we just pass them along
```
Comment (rhaschke) [inline moveit_ros/planning/plan_execution/src/plan_execution.cpp]: The comment isn't very helpful in the long run. Either drop it or replace it by a more generic one.

File: moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp [hunk 3]
```diff
@@ -115,6 +115,12 @@ class MoveItFakeControllerManager : public moveit_controller_manager::MoveItCont
           controllers_[name].reset(new InterpolatingController(name, joints, pub_));
         else
           ROS_ERROR_STREAM("Unknown fake controller type: " << type);
+
+        moveit_controller_manager::MoveItControllerManager::ControllerState state;
+        state.default_ = controller_list[i].hasMember("default") ? (bool)controller_list[i]["default"] : false;
+        state.active_ = true;
```
Comment (rhaschke) [inline moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp]: I'm not sure about the `active_` flag. Shouldn't this become true only if the controller is activated?

Comment (rhaschke) [reply moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp]: All the fake controllers are always active as they can be used without prior "activation".

File: moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp [hunk 4]
```diff
@@ -262,12 +268,9 @@ class MoveItFakeControllerManager : public moveit_controller_manager::MoveItCont
    * Controllers are all active and default.
    */
   moveit_controller_manager::MoveItControllerManager::ControllerState
-  getControllerState(const std::string& /*name*/) override
```
Comment (henningkayser) [inline moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp]: Shouldn't the same be implemented for the SimpleControllerManager?

Comment (rhaschke) [reply moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp]: Probably. @llach, could you provide the same implementation for SimpleControllerManager as well?
I suggest to split this PR into two separate commits then:
1. forwarding the controller set
2. fixing FakeControllerManager and SimpleControllerManager to consider the config `default`
So, Luca, please cleanup the commit history by appropriately squashing your changes.

**Comment by rhaschke** [comment]
@llach, could you please squash my fixup, so that we can rebase-merge this PR? Thanks.

**Comment by llach** [comment]
Done!

**Pull request merged**